monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunOnIslandButton.js

```diff
@@ -96,14 +97,54 @@ class RunOnIslandButton extends AuthComponent {
             this.setState({
                 showModal: true,
                 errorDetails: res['error_text'],
-                runningOnIslandState: MONKEY_STATES.FAILED
+                runningOnIslandState: MONKEY_STATES.FAILED,
+                errorMessage: this.getErrorMessageFromErrorText(res['error_text'])
               }
             );
           }
         }
       });
   }
 
+  getErrorMessageFromErrorText(errorText) {
+    if (errorText.includes('Permission denied:') || errorText.includes('Text file busy')) {
+      return this.getMonkeyAlreadyRunningContent()
+    } else if (errorText.startsWith('Copy file failed')) {
```

Comment:
Why is the screenshot showing "You encountered an undefined error" if it obviously should be "Some Monkey binaries are not found where they should be..."? Can we improve the error message selection? This is valuable information for the user, AV can update any time and delete the binary, we need to be sure to handle this case properly

Reply:
Good catch. Updated the message selection logic.
![image](https://user-images.githubusercontent.com/44770317/207619944-0b08a347-7284-457e-abbe-32541f97655b.png)